### PR TITLE
Add prow-github-action

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
This PR adds an action which replicates the prow bot functions using github actions.
This will support the following actions:
- Automatically merge PRs which has lgtm label. This action will scan for PRs with lgtm for every hour.
- Remove lgtm label if the PR is updated.
- Label issues with /kind, /area and /priority commands
The following commands are supported.
```
/assign 
/unassign 
/approve 
/retitle 
/area 
/kind 
/priority 
/remove 
/lgtm 
/close 
/reopen 
/lock 
/milestone 
/hold 
/cc 
/uncc
```
For more information, see https://github.com/jpmcb/prow-github-actions
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Add prow-github-action
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: .github/workflows/prow-pr-automerge.yml
Comment (spolti) [inline .github/workflows/prow-pr-automerge.yml:7]: Is there a way to trigger it after a comment or when the label is added? Like k8s prow?

Comment (sivanantha321) [reply .github/workflows/prow-pr-automerge.yml:7]: We can trigger the workflow when `lgtm` label is added to the PR by doing something like below.
```yaml
name: "Prow merge on lgtm label"
on:
  pull_request:
    types:
      - labeled

jobs:
  auto-merge:
    runs-on: ubuntu-latest
    if: contains(github.event.label.name, 'lgtm')
    steps:
      - uses: jpmcb/prow-github-actions@v1.1.3
        with:
          jobs: 'lgtm'
          github-token: "${{ secrets.GITHUB_TOKEN }}"
          merge-method: 'squash'
```
But there are limitations (that i know of).
1. The 'lgtm' label is added by the `github-actions` bot when we comment `/lgtm`. This won't trigger the workflow. This is a limitation of github. We can overcome this by using a `PAT Token` instead of `GITHUB_TOKEN`. You can read the detailed explanation here https://github.com/orgs/community/discussions/25565

2. Let's take a scenario where a PR is given `/lgtm` but not approved. Since `lgtm` label is added, the workflow will be triggered but the merge will fail as the PR is not approved. After you approved the PR, you have to remove the lgtm label and re add it to trigger the workflow.

File: .github/workflows/prow-pr-automerge.yml
Comment (terrytangyuan) [inline .github/workflows/prow-pr-automerge.yml:15]: Will this check whether the lgtm is given by reviewers in OWNERS file? In other words, can random people lgtm a PR and then it gets automatically merged?

Comment (sivanantha321) [reply .github/workflows/prow-pr-automerge.yml:15]: Only reviewers in the owner file can give lgtm. https://github.com/jpmcb/prow-github-actions/blob/main/docs/commands.md#owners

**Comment by terrytangyuan** [comment]
@sivanantha321 @yuzisun Do we want to merge this to try it out?

**Comment by spolti** [comment]
+1 to merge and start testing as the ci-prow is off for now.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**
